### PR TITLE
Adding wsclient retry capabilities

### PR DESCRIPTION
**Pull request by gwendolyngoetz**
### Description

Creates a new WSClient class that will retry on errors or handle the retry-after header for on throttled http 429 calls. Fully async future based.

Updates the mock web services to return a retry-after header when calling generates a 429 response.

Creates a class that can appropriately parse the value from the retry-after header as it can be either a date or number of seconds.

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)

- [x] Added the correct label (see [docs](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#adding-appropriate-labels) for more info): < feature | enhancement | bug | under-development | dependencies | infrastructure | ignore-for-release | database >
- [x] Assigned to a specific person, `civiform/developers`, or a [more specific round-robin list](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#adding-reviewers)
- [ ] Added an additional reviewer from outside your organization as FYI (if the primary reviewer is in the same organization as you)
- [x] Removed the release notes section if the title is sufficient for the release notes description, or put more details in that section.
- [x] Created unit and/or browser tests which fail without the change (if possible)
- [ ] Performed manual testing (Chrome and Firefox if it includes front-end changes)
- [ ] Extended the README / documentation, if necessary. For user-facing features, consider updating [the user docs](https://github.com/civiform/docs). For "under-the-hood" changes or things more relevant to developers, consider updating [the dev wiki](https://github.com/civiform/civiform/wiki).
- [ ] Ensured PII wasn't added to any new logs, unless it was guarded by `isDevOrStaging`
- [ ] Noted in the PR description which, if any, code in this PR was generated by AI.

Related to [#10555](https://github.com/civiform/civiform/issues/10555)

### CODE REVIEW

File: server/app/services/apibridge/RetryingWSClient.java [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    logger.debug(
+        "Retrying... URL={} Status={} Delay={}ms Retries={}",
+        wsResponse.getUri().toString(),
+        wsResponse.getStatus(),
+        retryDelayDuration.toMillis(),
+        retryCount);
+
+    CompletableFuture<WSResponse> retryFuture = new CompletableFuture<>();
+
+    var unused =
```
Comment (melanie-exygy) [inline server/app/services/apibridge/RetryingWSClient.java:151]: **question (non-blocking):** I'm curious why you need this unused variable.

File: server/test/services/apibridge/RetryAfterHeaderParserTest.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+import java.time.Duration;
+import java.time.Instant;
+import java.time.ZoneId;
+import org.junit.Test;
+
+public class RetryAfterHeaderParserTest {
+  private final Clock clock = Clock.fixed(Instant.parse("2015-10-21T07:28:00Z"), ZoneId.of("UTC"));
+
+  @Test
+  public void defaults_to_zero_when_argument_null_or_blank() {
```
Comment (melanie-exygy) [inline server/test/services/apibridge/RetryAfterHeaderParserTest.java:15]: **nitpick:** You could put the name of the method under test in the method name, although I realize there's only one method in the class right now.

File: server/test/services/apibridge/RetryAfterHeaderParserTest.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    var parser = new RetryAfterHeaderParser(clock);
+    assertThat(parser.parse(null)).isEqualTo(Duration.ZERO);
+    assertThat(parser.parse("")).isEqualTo(Duration.ZERO);
```
Comment (melanie-exygy) [inline server/test/services/apibridge/RetryAfterHeaderParserTest.java:18]: **suggestion (non-blocking):** Could it be possible for the header value to be a blank string?  Maybe worth a test assertion.

File: server/test/services/apibridge/RetryingWSClientTest.java [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    RetryingWSClient wsClient = spy(createWSClient(expectedHttpResponseCode));
+    String jsonBody = "{\"payload\": { \"accountNumber\": 1 }}";
+    WSResponse wsResponse = wsClient.post(POST_URL, jsonBody).toCompletableFuture().join();
+
+    assertExpectedMethodsCalled(wsClient, 1, 0);
+    assertThat(wsResponse.getStatus()).isEqualTo(expectedHttpResponseCode);
+  }
+
+  @Test
+  public void http_429_response_retries_using_retry_after_interval() {
```
Comment (melanie-exygy) [inline server/test/services/apibridge/RetryingWSClientTest.java:74]: **question (non-blocking):** I can't seem to find where you're setting the `retry-after` header for these test responses.  Can you direct me to where that happens?

File: server/test/services/apibridge/RetryingWSClientTest.java [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+    int expectedHttpResponseCode = 429;
+    RetryingWSClient wsClient = spy(createWSClient(expectedHttpResponseCode));
+    WSResponse wsResponse = wsClient.get(GET_URL).toCompletableFuture().join();
+
+    assertExpectedMethodsCalled(wsClient, 4, 3);
```
Comment (melanie-exygy) [inline server/test/services/apibridge/RetryingWSClientTest.java:82]: **suggestion (non-blocking):** I think it would be a good idea to test the scenario of a 429 response without the `retry-after` header since "Support for the Retry-After header on both clients and servers is still inconsistent."  (from https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Headers/Retry-After).

File: server/app/services/apibridge/RetryingWSClient.java (hunk 1 above)
Comment (gwendolyngoetz) [reply server/app/services/apibridge/RetryingWSClient.java:151]: Purely to make errorprone happy. :unamused:

File: server/test/services/apibridge/RetryAfterHeaderParserTest.java (hunk 3 above)
Comment (gwendolyngoetz) [reply server/test/services/apibridge/RetryAfterHeaderParserTest.java:18]: Not sure I follow, this test is checking for null or blank strings.

File: server/test/services/apibridge/RetryAfterHeaderParserTest.java (hunk 2 above)
Comment (gwendolyngoetz) [reply server/test/services/apibridge/RetryAfterHeaderParserTest.java:15]: I consider the test name to accurately describe what is going on

File: server/test/services/apibridge/RetryingWSClientTest.java (hunk 5 above)
Comment (gwendolyngoetz) [reply server/test/services/apibridge/RetryingWSClientTest.java:82]: I think the quality of the test is dubious as it has to excessively mock everything, but I added it below.

File: server/test/services/apibridge/RetryAfterHeaderParserTest.java (hunk 3 above)
Comment (melanie-exygy) [reply server/test/services/apibridge/RetryAfterHeaderParserTest.java:18]: I meant blank (vs. empty), so " ".

Comment (melanie-exygy) [reply server/test/services/apibridge/RetryAfterHeaderParserTest.java:18]: It's an edge case, so we can just go ahead without an extra test assertion.

Comment (gwendolyngoetz) [reply server/test/services/apibridge/RetryAfterHeaderParserTest.java:18]: If the value passed in is whitespace it would fail to parse as an int or date.
